res/res_usbradio.c

```diff
@@ -820,19 +858,108 @@ void ast_radio_print_rx_audio_stats(int fd, struct rxaudiostatistics *o)
 	}
 }
 
+/*!
+ * \brief Remove and free up all user devices.
+ */
+static void cleanup_user_devices(void)
+{
+	struct user_usb_device *device;
+	/* Remove all existing devices */
+	AST_RWLIST_WRLOCK(&user_devices);

```

Comment:
AST_LIST_TRAVERSE_SAFE_BEGIN is usually used when we want to remove a specific element by some property... it's not necessary here and might be a little simpler to just remove the first element in a loop, like this: https://github.com/asterisk/asterisk/blob/3dae2cf88fbe719a4f18d356cc77bb0be3fca7f5/main/srv.c#L239